File: regex/Python/regex4.py
import re

def rep( match ):
	name = match.group( 1 )
	if ( name == "name" ):
		return "Jack"
	if ( name == "cost" ):
		return "$20"
	return ""

print re.sub( "<(.*?)>", rep, "Dear <name>, You owe us <cost>." )

File: regex/Perl/regex4.pl
sub value($)
{
  my ( $key ) = @_;
  return 'Jack' if ( $key eq 'name' );
  return '$20' if ( $key eq 'cost' );
  return '';
}

$str = 'Dear <name>, You owe us <cost>';

$str =~ s/<(.*?)>/value($1)/eg;

print "$str\n";
